File: common/src/main/java/com/hartwig/actin/molecular/sort/CopyNumberComparator.java
package com.hartwig.actin.molecular.sort;

import java.util.Comparator;

import com.hartwig.actin.molecular.datamodel.driver.CopyNumberDriver;

import org.jetbrains.annotations.NotNull;

public class CopyNumberComparator implements Comparator<CopyNumberDriver> {

    @Override
    public int compare(@NotNull CopyNumberDriver copyNumberDriver1, @NotNull CopyNumberDriver copyNumberDriver2) {
        return copyNumberDriver1.gene().compareTo(copyNumberDriver2.gene());
    }
}

File: datamodel/src/main/kotlin/com/hartwig/actin/datamodel/molecular/sort/driver/CopyNumberComparator.kt
package com.hartwig.actin.datamodel.molecular.sort.driver

import com.hartwig.actin.datamodel.molecular.driver.CopyNumber

class CopyNumberComparator : Comparator<CopyNumber> {

    private val comparator = Comparator.comparing<CopyNumber, CopyNumber>({ it }, DriverComparator())
        .thenComparing({ it }, GeneAlterationComparator())
    
    override fun compare(copyNumber1: CopyNumber, copyNumber2: CopyNumber): Int {
        return comparator.compare(copyNumber1, copyNumber2)
    }
}
